### PR TITLE
Checks if currency is available in list

### PR DESCRIPTION
**Pull request by lomefin**
Solves issue #48 

Checks if currency is available in the currency list, throws error if not found.

### CODE REVIEW

File: lib/eu_central_bank.rb [hunk 1]
```diff
@@ -67,6 +69,7 @@ def exchange_with(from, to_currency, date=nil)
   end
 
   def get_rate(from, to, date = nil)
+    check_currency_available(from) and check_currency_available(to)
```
Comment (antstorm) [inline lib/eu_central_bank.rb]: No need for `and` here, just place method calls on separate lines

File: lib/eu_central_bank.rb [hunk 2]
```diff
@@ -136,6 +139,12 @@ def import_rates(format, s, opts = {})
     self
   end
 
+  def check_currency_available(currency)
+    return true if currency == "EUR"
+    return true if CURRENCIES.include?(currency)
+    raise CurrencyUnavailable
```
Comment (antstorm) [inline lib/eu_central_bank.rb]: You definitely wanna give it an error message with currency that is unavailable, like `"No rates available for #{currency}"`, which makes it much more clear what went wrong

File: lib/eu_central_bank.rb [hunk 3]
```diff
@@ -136,6 +139,12 @@ def import_rates(format, s, opts = {})
     self
   end
 
+  def check_currency_available(currency)
+    return true if currency == "EUR"
+    return true if CURRENCIES.include?(currency)
```
Comment (antstorm) [inline lib/eu_central_bank.rb]: I'd suggest `currency.to_s` in both lines above, since somebody might try to pass a symbol

**Comment by antstorm** [comment]
Couple of comments, but a good change, thank you! With a few corrections, I'm happy to merge it, great job!

File: lib/eu_central_bank.rb (hunk 1 above)
Comment (lomefin) [inline lib/eu_central_bank.rb]: Yeah, at the beginning the detection was there and then I liked to have a one-liner.

File: lib/eu_central_bank.rb (hunk 2 above)
Comment (lomefin) [inline lib/eu_central_bank.rb]: Great comment!

File: lib/eu_central_bank.rb [hunk 4]
```diff
@@ -136,6 +147,13 @@ def import_rates(format, s, opts = {})
     self
   end
 
+  def check_currency_available(currency)
+    currency_string = currency.to_s
+    return true if currency_string == "EUR"
+    return true if CURRENCIES.include?(currency_string)
+    raise CurrencyUnavailable.new(currency_string)
```
Comment (antstorm) [inline lib/eu_central_bank.rb]: Yes, that's better, but I don't think there's any need for additional behaviour (like `initialize` and `to_s` methods) apart from:

```ruby
raise CurrencyUnavailable, "No rates available for #{currency_string}"
```

You might wanna use that when you're returning more than just an error string, which is not the case here

Comment (lomefin) [inline lib/eu_central_bank.rb]: Got it, I based that code on a proposal I saw the other day. I'm not very used to use exceptions in Ruby :)

**Comment by antstorm** [comment]
@lomefin perfect! Can you please squash the commits into a single one and I'll merge it.

**Comment by antstorm** [comment]
Thank you!

**Pull request merged**